### PR TITLE
Handle computing of nearest neighbor metrics for units with few spikes

### PR DESCRIPTION
**Pull request by khl02007**
* Add a new parameter to `nearest_neighbors_isolation` and `nearest_neighbors_noise_overlap` functions called `min_spikes_for_nn`
* Units with fewer spikes than `min_spikes_for_nn` (small clusters) are given `np.NaN` as the metric and a warning is printed
* `min_spikes_for_nn` is set to 10 by default
* Metrics are computed for units with more spikes than `min_spikes_for_nn` but after first excluding the small clusters (since nearest neighbor metrics do pairwise comparison across all clusters)

### CODE REVIEW

**Comment by khl02007** [comment]
should solve #589

**Comment by khl02007** [comment]
looks like the test fails because the data used for test is a sorting with multiple segments and I don't specify the segment when I get the number of spikes for all units. Not sure what the best thing to do is. Should I change my PR to first concatenate the sorting?

**Comment by samuelgarcia** [comment]
Hi Kyu,
for me `min_spikes_for_nn` should be global  and so the of spike per unit sum across segment)
What do you think ?

File: spikeinterface/toolkit/qualitymetrics/pca_metrics.py [hunk 1]
```diff
@@ -333,64 +344,82 @@ def nearest_neighbors_isolation(waveform_extractor: si.WaveformExtractor, this_u
...
-
-    # if no unit is within neighborhood of target unit, then just say isolation is 1 (best possible)
-    if not other_units_ids:
-        nn_isolation = 1
-
-    # if there are units to compare, then compute isolation with each
+    
+    sorting = waveform_extractor.sorting
+    all_units_ids = sorting.get_unit_ids()
+    n_spikes_all_units = compute_num_spikes(waveform_extractor)
```
Comment (alejoe91) [inline spikeinterface/toolkit/qualitymetrics/pca_metrics.py:350]: @samuelgarcia I ended up using the `compute_num_spikes` from the `quality_metrics` module. Will make a separate PR for the core function

**Comment by samuelgarcia** [comment]
THanks.
